openmp/libomptarget/test/offloading/fortran/basic-target-parallel-do.f90

```diff
@@ -0,0 +1,31 @@
+! Basic offloading test with a target region
+! REQUIRES: flang
+! UNSUPPORTED: nvptx64-nvidia-cuda-LTO
+! UNSUPPORTED: aarch64-unknown-linux-gnu
+! UNSUPPORTED: aarch64-unknown-linux-gnu-LTO
+! UNSUPPORTED: x86_64-pc-linux-gnu
+! UNSUPPORTED: x86_64-pc-linux-gnu-LTO
+
+! RUN: %libomptarget-compile-fortran-run-and-check-generic
+program main
+   use omp_lib
+   integer :: x(100)
+   integer :: errors = 0
+   integer :: i
+
+   !$omp target parallel do map(from: x)
+   do i = 1, 100
+       x(i) = i
+   end do
+   !$omp end target parallel do
```

Comment:
Since, AFAIU, offloading can silently fallback to the host, should we, in our offloading tests, always verify that offloading did indeed succeed somehow? I did this [previously](https://github.com/ergawy/llvm-project/blob/794bf300693e7826a7e2b7d50e6823d9dd421d4e/openmp/libomptarget/test/offloading/fortran/target_update.f90#L45) by taking device IDs and comparing them to make sure they are different. However, I am not sure this is the best approach.

Or do we not need to verify that at all?